### PR TITLE
test(sdk): validate handle_response status and length edge cases

### PR DESCRIPTION
**Pull request by ankushT369**
## Rationale

Tests for `handle_response` did not cover important edge cases around status codes and response length, leaving potential failure scenarios unverified.

## What changed?

`handle_response` behavior was not validated for error conditions such as non-zero status codes or when the declared response length exceeds available data.

Added unit tests covering both success and failure paths, including boundary conditions for status and length handling.

## AI Usage

None

### CODE REVIEW

**Comment by ankushT369** [comment]
@hubcio Hi can you please review this ?

File: core/sdk/src/tcp/tcp_client.rs [hunk 1]
```diff
@@ -884,4 +901,32 @@ mod tests {
...
+        let mut stream = make_dummy_stream(&[1u8; 10]).await;
+        let tcp_client = TcpClient::handle_response(0, 0, &mut stream).await;
+        assert!(tcp_client.is_ok());
+    }
+
+    #[tokio::test]
+    async fn should_return_ok_when_length_is_less_than_data() {
+        let mut stream = make_dummy_stream(&[1u8; 10]).await;
+        let tcp_client = TcpClient::handle_response(0, 5, &mut stream).await;
+        assert!(tcp_client.is_ok());
```
Comment (hubcio) [inline core/sdk/src/tcp/tcp_client.rs:923]: missing `length == 1` boundary. guard is `if length <= 1`. tests cover 0 and 5; 1 is the inclusive edge, off-by-one risk if guard becomes `<`. add `handle_response(0, 1, ...) == Ok(Bytes::new())`.

**Review by hubcio — CHANGES_REQUESTED** [review]

**Comment by ankushT369** [comment]
@hubcio Hey I didnt get the **Check status** failing in the previous commit

**Review by hubcio — APPROVED** [review]

**Review by spetz — APPROVED** [review]

**Review by numinnex — APPROVED** [review]
